pyriemann/utils/mean.py

```diff
@@ -30,52 +30,60 @@ def mean_ale(covmats, tol=10e-7, maxiter=50, sample_weight=None):
 
     Returns
     -------
-    C : ndarray, shape (n_channels, n_channels)
+    C : ndarray, shape (n, n)
         ALE mean.
 
     Notes
     -----
     .. versionadded:: 0.2.4
 
+    See Also
+    --------
+    mean_covariance
+
     References
     ----------
     .. [1] `Approximate Joint Diagonalization and Geometric Mean of Symmetric
         Positive Definite Matrices
         <https://arxiv.org/abs/1505.07343>`_
         M. Congedo, B. Afsari, A. Barachant, M. Moakher. PLOS ONE, 2015
     """
-    n_matrices, n_channels, _ = covmats.shape
+    n_matrices, n, _ = covmats.shape
     sample_weight = check_weights(sample_weight, n_matrices)
 
     # init with AJD
-    B, _ = ajd_pham(covmats)
+    B = ajd_pham(covmats)[0]
 
     crit = np.inf
     for _ in range(maxiter):
-        J = np.einsum('a,abc->bc', sample_weight, logm(B.T @ covmats @ B))
-        update = np.diag(np.diag(expm(J)))
-        B = B @ invsqrtm(update)
+        J = np.einsum(
+            'a,abc->bc',
+            sample_weight,
+            logm(B @ covmats @ B.conj().T)
+        )
+        delta = np.real(np.diag(expm(J)))
+        B = (np.abs(delta) ** -.5)[:, np.newaxis] * B
 
-        crit = distance_riemann(np.eye(n_channels), update)
+        crit = distance_riemann(np.eye(n), np.diag(delta))
         if crit <= tol:
             break
     else:
-        warnings.warn('Convergence not reached')
+        warnings.warn("Convergence not reached")
 
+    J = np.einsum('a,abc->bc', sample_weight, logm(B @ covmats @ B.conj().T))
     A = np.linalg.inv(B)
-    J = np.einsum('a,abc->bc', sample_weight, logm(B.T @ covmats @ B))
-    C = A.T @ expm(J) @ A
+    C = A @ expm(J) @ A.conj().T
     return C
 
 
 def mean_alm(covmats, tol=1e-14, maxiter=100, sample_weight=None):
-    r"""Ando-Li-Mathias (ALM) mean of SPD matrices.
+    r"""Ando-Li-Mathias (ALM) mean of SPD/HPD matrices.
 
     Return the geometric mean recursively [1]_, generalizing from:
 
     .. math::
-        \mathbf{C} = A^{\frac{1}{2}}(A^{-\frac{1}{2}}B^{\frac{1}{2}}
-                     A^{-\frac{1}{2}})^{\frac{1}{2}}A^{\frac{1}{2}}
+        \mathbf{C} = X_1^{\frac{1}{2}} (X_1^{-\frac{1}{2}}X_2^{\frac{1}{2}}
+                     X_1^{-\frac{1}{2}})^{\frac{1}{2}} X_1^{\frac{1}{2}}
```

Comment:
we should make consistent how we document the mean equations. Here you only have 2 matrices but one can have n_matrices > 2.

Reply:
I am afraid that recursive nature of this mean prevents to write an analytic formula.
@sylvchev , what do you think?

Reply:
By the way, the link http://bezout.dm.unipi.it/software/mmtoolbox/ does not exist.

Reply:
The ALM is useful for a sanity check as it is a reference in geometry but this is not scale due to the recursion, so this is a special case.